dashboard/app/assets/javascripts/levels/contract_match.js.jsx

```diff
@@ -0,0 +1,257 @@
+    onDomainRemove: function (domainKey) {
+      this.setState({
+        domainTypes:
+          $.grep(this.state.domainTypes, function (object) {
+            return object.key !== domainKey;
+          })
+      });
+    },
+    render: function () {
+      return (
+        <div>
+          <div id='sectionTitle'>Name</div>
+          <div>
+            <input id='functionNameText' onChange={this.onNameChangeEvent} placeholder='Name' type='text' value={this.state.name}/>
+          </div>
+          <div id='sectionTitle'>Domain</div>
+          <DomainsList
+            domainTypes={this.state.domainTypes}
+            onDomainChange={this.onDomainChange}
+            onDomainAdd={this.onDomainAdd}
+            onDomainRemove={this.onDomainRemove}/>
+          <div id='sectionTitle'>Range</div>
+          <TypeChooser type={this.state.rangeType} onTypeChange={this.onRangeChange}/>
+        </div>
+      )
+    }
+  });
+
+  var DomainsList = React.createClass({
+    render: function () {
+      var self = this;
+      var sortedDomains = this.props.domainTypes.sort(function (a,b) {
+        return a.order > b.order;
+      });
+      var lastNode = this.props.domainTypes[this.props.domainTypes.length - 1];
+      var typeChoiceNodes = $.map(sortedDomains, function (object) {
+        var isLastNode = (object === lastNode);
+        return (
+          <div style={isLastNode ? {float: 'left'} : {}}>
+            <TypeChooser order={object.order} type={object.type} key={object.key}
+              onTypeChange={curry(self.props.onDomainChange, object.key)}/>
+            <button onClick={curry(self.props.onDomainRemove, object.key)}>x</button>
+          </div>
+        );
+      });
+      return (
+        <div className="domainsList">
+          {typeChoiceNodes}
+          <button onClick={this.props.onDomainAdd}>Add Domain</button>
+        </div>
+      )
+    }
+  });
+
+  var TypeChooser = React.createClass({
+    handleChange: function(event) {
+      this.props.onTypeChange(event.target.value);
+    },
+    render: function () {
+      var divStyle = {
+        backgroundColor: typesToColors[this.props.type]
+      };
+      return (
+        <select value={this.props.type} onChange={this.handleChange} style={divStyle}>
+          <option value={blockValueType.NONE} disabled style={{display: 'none'}}>Choose a Type</option>
+          <option value={blockValueType.NUMBER}>{blockValueType.NUMBER}</option>
+          <option value={blockValueType.STRING}he>{blockValueType.STRING}</option>
+          <option value={blockValueType.IMAGE}>{blockValueType.IMAGE}</option>
+          <option value={blockValueType.BOOLEAN}>{blockValueType.BOOLEAN}</option>
+        </select>
+      )
+    }
+  });
+
+  var contractForm = React.render(<ContractForm />, document.getElementById('contractForm'));
+
+  /**
+   * @param {ContractForm} contractForm
+   * @param {Object} levelData
+   */
+  window.getResult = function (contractForm, levelData) {
```

Comment:
Right now the partial `dashboard/app/assets/javascripts/sendReport.js.erb` (which provides the "Submit" button & does the AJAX calls for these assessments) looks for a method on the top-level named `getResult`. Wondering what might be a cleaner way of passing this method along